### PR TITLE
Fixed getBaseDirectoryInclude when one directory name include the oth…

### LINKED ISSUE #2230: ChangeWatcher doesn't trigger onchange when one directory name include the other
### Description

ChangeWatch doesn't trigger onchange when one directroy name include the other
### Repro steps

Please run `fake build` to see repro steps https://gist.github.com/humhei/1c8a8489de9e0ade52f6509e0d898f66
```fsharp
open Fake.Core
open Fake.IO
open Fake.IO.Globbing

let file1 = Path.getFullName "TestLib/Test.fs"
let file2 = Path.getFullName "TestLib2/Test.fs"
let pattern =
    let files = [file1;file2]
    { BaseDirectory = Path.getFullName "./"
      Includes = files
      Excludes = [] }

ChangeWatcher.run (fun changes ->
    changes |> Seq.iter (fun change -> printfn "file %s changed" change.FullPath)
) pattern


```

### Expected behavior
```fsharp
File.append file1 ["\n"] //this will trigger onchange
File.append file2 ["\n"] //this will trigger onchange
```
### Actual behavior
```fsharp
File.append file1 ["\n"] //this will trigger onchange
File.append file2 ["\n"] //this will not trigger onchange
```

### Known workarounds
Carfully naming directorys 
Or
change https://github.com/fsharp/FAKE/blob/master/src/app/Fake.IO.FileSystem/GlobbingFileSystem.fs#L147
 ```fsharp
|> Seq.exists (fun p -> d.StartsWith p && p <> d)
 ```
    to 
```fsharp
|> Seq.exists (fun p -> d.StartsWith (p + "\\") && p <> d)
```

### Related information 

* win10
* FAKE 5.11.1

### PR DESCRIPTION
**Pull request by humhei**
Fixes #2230

### CODE REVIEW

**Comment by humhei** [comment]
I don't know why CI sucessed in windows but failed in linux

**Comment by matthid** [comment]
Maybe the change has broken something on unix (note that on unix path separator is / not \)
I can take a look but it will take a couple of days.

It might also be that the existing test has a bug?

File: src/app/Fake.IO.FileSystem/GlobbingFileSystem.fs [hunk 1]
```diff
@@ -144,7 +144,7 @@ module GlobbingPattern =
             directoryIncludes
             |> Seq.filter (fun d ->
                             directoryIncludes
-                            |> Seq.exists (fun p -> d.StartsWith p && p <> d)
+                            |> Seq.exists (fun p -> d.StartsWith (p + "\\") && p <> d)
```
Comment (matthid) [inline src/app/Fake.IO.FileSystem/GlobbingFileSystem.fs]:
```suggestion
                            |> Seq.exists (fun p -> d.StartsWith (p + string System.IO.Path.DirectorySeparatorChar) && p <> d)
```

Maybe this helps to make it work on other platforms

**Comment by matthid** [comment]
Huh, it looks like it worked, thanks for the fix and the test!

**Pull request merged**